### PR TITLE
Enable coverage analysis on codecov.io & add badge

### PR DESCRIPTION
**Pull request by FarnazH**
I followed theochem/iodata to add coverage.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,6 +22,12 @@ osx_image: xcode8.3
 dist: trusty
 sudo: false
 
+# Only run tests on master branch to avoid wasting Travis resources.
+# PRs will still be tested.
+branches:
+  only:
+    - master
+
```
Comment (tovrstra) [inline .travis.yml]: I'v recently update the [`.travis.yml`](https://github.com/theochem/python-cython-ci-example/blob/master/.travis.yml) file of theochem/python-cython-ci-example to include also branches that match `/^[0-9]+\.[0-9]+(\.[0-9]+)?([ab][0-9]+)?$/`, to activate travis also for tagged releases.

**Review by tovrstra — APPROVED** [review]

**Comment by FarnazH** [comment]
@tovrstra I just added that as well. Is it true that coverage checks test files as well? If so, is that what we want?

**Comment by tovrstra** [comment]
That's right. It is useful. Sometimes, due to mistakes, tests may not get execute, while in practice they should. The simplest example, which I've had a few times due to copy-paste mistakes, is that two tests have the same name. In that case, only the last one gets executed and coverage will detect it.

The downside is that you have to pay attention when you raise a [SkipTest](http://nose.readthedocs.io/en/latest/plugins/skip.html) exception. Then you also need to add a special comment to tell coverage to ignore the test too. See http://coverage.readthedocs.io/en/coverage-4.2/excluding.html

**Comment by tovrstra** [comment]
Looking good. Merging...

**Pull request merged**